### ISSUE TITLE
mini.surround - visual mode surround on block puts opening '(),{},[]' on line before and closing on line after.

### ISSUE THREAD

**Issue by vsandstrom** [echasnovski/mini.nvim#150]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)

### Module(s)

mini.surround

### Description

I would like to be able to select a block in visual mode and surround them with parenthesis or braces, and it should do a linebreak after the opening symbol and another after the block before closing. 

It should look like this afterwards.

```c
(
    printf("hello");
)
```

instead of this.

```c
(printf("hello");)
```

**Comment by echasnovski** [comment]
That is mostly by design. Main problem here is the formatting afterwards: I don't think there is a universal, filetype-agnostic rule to do this. The closest one seems to be "indent all lines by a single tab or tab worth of spaces", but it is still not universal (and not so easy to implement at the moment).

I'll have this in mind.

**Comment by xigoi** [comment]
I think indenting all lines by one `shiftwidth` would definitely be the most useful behavior.

**Comment by echasnovski** [comment]
So I finally gave up on not adding this, since I managed to find a quite manageable implementation of this feature. In particular, I didn't know that indent/dedent can be conveniently done programmatically with `:[range] >` and `:[range] <` (see `:h :>`).

This functionality is not default and is activated with new `respect_selection_type = true` option. It changes how linewise and blockwise addition works along with deletion that looks like it should revert linewise add. I hated introducing new option for this, but I can see that this behavior is useful and breaking changes are bad.

See details [here](https://github.com/echasnovski/mini.nvim/blob/main/doc/mini-surround.txt#L555).
